src/metabase/db/liquibase.clj

```diff
@@ -181,20 +189,44 @@
   see https://github.com/metabase/metabase/issues/3715"
   [db-type :- s/Keyword
    conn    :- java.sql.Connection]
-  (let [liquibase-table-name (if (#{:h2 :mysql} db-type)
-                               "DATABASECHANGELOG"
-                               "databasechangelog")
+  (let [liquibase-table-name (changelog-table-name db-type)
         fresh-install?       (jdbc/with-db-metadata [meta {:connection conn}] ; don't migrate on fresh install
                                (empty? (jdbc/metadata-query
                                         (.getTables meta nil nil liquibase-table-name (u/varargs String ["TABLE"])))))
         statement            (format "UPDATE %s SET FILENAME = ?" liquibase-table-name)]
     (when-not fresh-install?
       (jdbc/execute! {:connection conn} [statement "migrations/000_migrations.yaml"]))))
 
-(defn rollback-one
-  "Roll back the last migration."
-  [^Liquibase liquibase]
-  (.rollback liquibase 1 ""))
+(defn- extract-numbers
+  "Returns contiguous integers parsed from string s"
+  [s]
+  (map #(Integer/parseInt %) (re-seq #"\d+" s)))
+
+(defn- current-major-version
+  "Returns the major version of the running Metabase JAR"
+  []
+  (second (extract-numbers (:tag config/mb-version-info))))
+
+(defn rollback-major-version
+  "Roll back migrations later than given Metabase major version"
+  ;; default rollback to previous version
+  ([db-type conn liquibase]
+   ;; get current major version of Metabase we are running
+   (rollback-major-version db-type conn liquibase (dec (current-major-version))))
+
+  ;; with explicit target version
+  ([db-type conn ^Liquibase liquibase target-version]
```

Comment:
if we run this with `java -jar` are there any situations where `target-version` can come in as a string and need to be parsed?

Reply:
No, the target version is not an argument to the command at this point.